src/common/docker-compose/index.ts

```diff
@@ -87,6 +94,37 @@ export const generate = (dependency_manager: DependencyManager): DockerComposeTe
     }
   });
 
+  const expanded_params = dotenvExpand({ parsed: env_params_to_expand }).parsed;
+  dependency_manager.graph.nodes.forEach(node => {
```

Comment:
We're doing something wrong if we're iterating over the nodes twice when we generate the docker compose. Upon complete generation, the graph of nodes and edges should contain all the details needed to do this on a single pass.